drivers/gpio/gpio_davinci.c

```diff
@@ -64,10 +62,24 @@ struct gpio_davinci_config {
 	const struct pinctrl_dev_config *pcfg;
 };
 
+const unsigned int offset_array[5] = {0x10, 0x38, 0x60, 0x88, 0xb0};
+#define MAX_REGS_BANK ARRAY_SIZE(offset_array)
+
+static struct gpio_davinci_regs *gpio_davinci_get_regs(const struct device *dev, uint8_t bank)
```

Comment:
What's your plan on how to calculate this bank value?

Reply:
So, the bank value is basically `pin / 32`.

```c
static uint32_t __gpio_mask(gpio_pin_t gpio)
{
	return BIT(gpio % 32);
}

static uint8_t __gpio_bank(gpio_pin_t gpio) {
	return gpio / 32;
}
```

Reply:
I think passing the pin instead of bank and calculating the bank number would make that function more suitable for future support for multiple banks.

Reply:
Passing a single pin should work fine, but cannot be used to set multiple pins at once for a bank like it is done right now for BANK0